### PR TITLE
Adding generators

### PR DESCRIPTION
**Pull request by scottrobertson**
This will automatically add the route, and generate a config file.

Screenshot 2021-01-17 at 20 33 37@2x

### CODE REVIEW

File: lib/generators/tokenable/verifier_generator.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'rails/generators/active_record'
+
+module Tokenable
```
Comment (marcqualie) [inline lib/generators/tokenable/verifier_generator.rb:5]: Should the path be `lib/tokenable/generators` to match the file name path?

Comment (scottrobertson) [reply lib/generators/tokenable/verifier_generator.rb:5]: Na, it needs the sub folder for the namespace. Will play around and see if i can get it to work without.

File: lib/generators/templates/verifier_migration.rb.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddTokenableVerifierTo<%= table_name.camelize %> < ActiveRecord::Migration<%= migration_version %>
+  def change
+    add_column :<%= table_name %>, :tokenable_verifier, :uuid
```
Comment (marcqualie) [inline lib/generators/templates/verifier_migration.rb.erb:3]:
```suggestion
    add_column :<%= table_name %>, :tokenable_verifier, :string
```

File: lib/generators/templates/tokenable.rb.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Tokenable::Config.lifespan = nil
```
Comment (marcqualie) [inline lib/generators/templates/tokenable.rb.erb:1]: Should we add a sensible default for expiry? Even something like 7 days would be better than infinite

Comment (scottrobertson) [reply lib/generators/templates/tokenable.rb.erb:1]: Yeah i was going to discuss that with you sometime. It feels like it needs a default.

File: lib/generators/tokenable/verifier_generator.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  module Generators
+    class VerifierGenerator < ActiveRecord::Generators::Base
+      source_root File.expand_path('../templates', __dir__)
+
+      def generate_model
+        invoke 'active_record:model', [name], migration: false unless model_exists? && behavior == :invoke
+      end
+
+      def add_to_model
+        inject_into_file "app/models/#{file_name}.rb", "  include Tokenable::Verifier\n", after: " < ApplicationRecord\n"
```
[the remark this answers is not in the document]
Comment (scottrobertson) [reply lib/generators/tokenable/verifier_generator.rb:15]: It really is!

File: lib/generators/templates/tokenable.rb.erb (hunk 3 above)
Comment (scottrobertson) [reply lib/generators/templates/tokenable.rb.erb:1]: Will add in another PR.

**Pull request merged**